PerformanceCalculator/Profile/ProfileProcessor.cs

```diff
@@ -0,0 +1,203 @@
+                        client.DownloadFile("https://osu.ppy.sh/osu/" + playData[i].beatmap_id, path);
+                    }
+                    catch
+                    {
+                        Console.WriteLine("Error in Downloading Beatmaps");
+                    }
+                }
+
+                var workingBeatmap = new ProcessorWorkingBeatmap(path);
+
+                //Stats Calculation
+                var ruleset = new OsuRuleset();
+
+                double countmiss = playData[i].countmiss;
+                double count50 = playData[i].count50;
+                double count100 = playData[i].count100;
+                double count300 = playData[i].count300;
+
+                double accuracy = (count50 + 2 * count100 + 6 * count300) / (6 * (countmiss + count50 + count100 + count300));
+                var maxCombo = (int)playData[i].maxcombo;
+
+                var statistics = new Dictionary<HitResult, object>
+                {
+                    {HitResult.Great, (int)count300},
+                    {HitResult.Good, (int)count100},
+                    {HitResult.Meh, (int)count50},
+                    {HitResult.Miss, (int)countmiss}
+                };
+
+                //mods are in a bitwise binary sum, with each mod given by the enum Mods
+                List<Mod> mods = new List<Mod>();
+                int enabledMods = (int)playData[i].enabled_mods;
+
+                if(((int)Mods.Hidden & enabledMods) != 0)
+                    mods.Add(new OsuModHidden());
+
+                if(((int)Mods.Flashlight & enabledMods) != 0)
+                    mods.Add(new OsuModFlashlight());
+
+                //no need to include Nightcore because all nightcore maps also return DoubleTime by default
+                if(((int)Mods.DoubleTime & enabledMods) != 0)
+                    mods.Add(new OsuModDoubleTime());
+
+                if(((int)Mods.HalfTime & enabledMods) != 0)
+                    mods.Add(new OsuModHalfTime());
+
+                if(((int)Mods.HardRock & enabledMods) != 0)
+                    mods.Add(new OsuModHardRock());
+
+                if(((int)Mods.Easy & enabledMods) != 0)
+                    mods.Add(new OsuModEasy());
+
+                if(((int)Mods.NoFail & enabledMods) != 0)
+                    mods.Add(new OsuModNoFail());
+
+                if(((int)Mods.SpunOut & enabledMods) != 0)
+                    mods.Add(new OsuModSpunOut());
+
+                Mod[] finalMods = mods.ToArray();
+
+                var scoreInfo = new ScoreInfo
+                {
+                    Accuracy = accuracy,
+                    MaxCombo = maxCombo,
+                    Mods = finalMods,
+                    Statistics = statistics
+                };
+
+                workingBeatmap.Mods.Value = finalMods;
+
+                var categoryAttribs = new Dictionary<string, double>();
+                double pp = ruleset.CreatePerformanceCalculator(workingBeatmap, scoreInfo).Calculate(categoryAttribs);
```

Comment:
`Calculate` can be called without any arguments so if you're not using `categoryAttribs` anywhere you remove it.
```suggestion
                double pp = ruleset.CreatePerformanceCalculator(workingBeatmap, scoreInfo).Calculate();
```